cpp/tests/utilities/column_wrapper.cuh

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef COLUMN_WRAPPER_H
+#define COLUMN_WRAPPER_H
+
+#include "cudf.h"
+#include "cudf_test_utils.cuh"
+#include "rmm/rmm.h"
+#include "rmm/thrust_rmm_allocator.h"
+#include "utilities/bit_util.cuh"
+#include "utilities/type_dispatcher.hpp"
+
+#include <thrust/equal.h>
+#include <thrust/logical.h>
+#include <bitset>
+#include <string>
+
+#ifndef CUDA_RT_CALL
+#define CUDA_RT_CALL(call)                                                    \
+  do {                                                                        \
+    cudaError_t cudaStatus = (call);                                          \
+    if (cudaSuccess != cudaStatus) {                                          \
+      fprintf(stderr,                                                         \
+              "ERROR: CUDA RT call \"%s\" in line %d of file %s failed with " \
+              "%s (%d).\n",                                                   \
+              #call, __LINE__, __FILE__, cudaGetErrorString(cudaStatus),      \
+              cudaStatus);                                                    \
+      exit(1);                                                                \
+    }                                                                         \
+  } while (0)
+#endif
+
+namespace cudf {
+namespace test {
+
+/**---------------------------------------------------------------------------*
+ * @brief Wrapper for a gdf_column used for unit testing.
+ *
+ * An abstraction on top of a gdf_column that provides functionality for
+ * allocating, initializing, and otherwise managing gdf_columns for passing to
+ * libcudf APIs in unit testing.
+ *
+ * @tparam ColumnType The underlying data type of the column
+ *---------------------------------------------------------------------------**/
+template <typename ColumnType>
```

Comment:
1. No need to append the word "Type" in this context IMHO.
2. It's the element type, not the column type.

So: `template <typename Element>` . Actually, I wouldn't mind `template <typename E>`.